File: src/main/java/core/datatype/CBItem.java
package core.datatype;

/**
 * @author thomas.werth
 * @version
 */
public class CBItem implements ComboItem {

	private String m_sText;
	private int m_iId;

	public CBItem(String text, int id) {
		m_sText = text;
		m_iId = id;
	}

	@Override
	public final int getId() {
		return m_iId;
	}

	@Override
	public final String getText() {
		return m_sText;
	}

	@Override
	public final boolean equals(Object obj) {
		if (obj instanceof CBItem) {
			final CBItem temp = (CBItem) obj;

			if (this.getId() == temp.getId()) {
				return true;
			}
		}
		return false;
	}

	@Override
	public final String toString() {
		return m_sText;
	}
}

File: src/main/java/core/datatype/CBItem.kt
package core.datatype

/**
 * Combo item that associates an ID to a String.
 *
 * @author thomas.werth
 */
open class CBItem(override val text: String, override val id: Int) : ComboItem {

    override fun equals(other: Any?): Boolean {
        return if (other is CBItem) {
            id == other.id
        } else false
    }

    override fun toString(): String {
        return text
    }

    override fun hashCode(): Int {
        var result = text.hashCode()
        result = 31 * result + id
        return result
    }
}
